enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,481 @@
+spec:
+ nodeSelector:
+   nodeSelectorTerms:
+   - matchExpressions:
+     - key: kubernetes.io/hostname
+       operator: In
+       values:
+       - worker-0
+       - worker-1
+ tolerations:
+Status:
+ Phase: Discovered
+ NodeDeviceMap:
+   Worker-0:
+     blockdevices:
+     - name: sda
+       model: SAMSUNG MZ7LN512
+       vendor: 'ATA'
+       kname: sda
+       pkname:
+       type: disk
+       uuid:
+       tran: sata
+       ro: '0'
+       parttype:
+       serial: S1ZKNXAH103764
+       size: 477G
+       rota: '0'
+       mountpoint:
+       children:
+       - name: sda1
+         model:
+         vendor:
+         kname: sda1
+         pkname: sda
+         type: part
+         uuid: 5ef9311f-843c-4781-b3d5-cae9cbd61afb
+         tran:
+         ro: '0'
+         parttype: '0x83'
+         serial:
+         size: 1G
+         rota: '0'
+         mountpoint: "/boot"
+       - name: sda2
+         model:
+         vendor:
+         kname: sda2
+         pkname: sda
+         type: part
+         uuid: M1Mtir-v51o-Chsr-5zY1-kvD2-JCn7-IWx5AF
+         tran:
+         ro: '0'
+         parttype: '0x8e'
+         serial:
+         size: 476G
+         rota: '0'
+         mountpoint:
+         children:
+         - name: fedora_dhcp35--228-root
+           model:
+           vendor:
+           kname: dm-0
+           pkname: sda2
+           type: lvm
+           uuid: 3301bfbf-7372-4bc4-80d1-183255352628
+           tran:
+           ro: '0'
+           parttype:
+           serial:
+           size: 50G
+           rota: '0'
+           mountpoint: "/"
+   Worker-1:
+     blockdevices:
+     - name: sda
+       model: SAMSUNG MZ7LN512
+       vendor: 'ATA     '
+       kname: sda
+       pkname:
+       type: disk
+       uuid:
+       tran: sata
+       ro: '0'
+       parttype:
+       serial: S1ZKNXAH103764
+       size: 477G
+       rota: '0'
+       mountpoint:
+```
+Now, once we have the detected devices the administrator can create the localPVs either by explicitly selecting the disks. This can be done by the `localVolume` CR. The other option would be to create localPVs via the AutoDetectVolume CR and passing the inclusion filters in it.
+
+#### Workflow of LocalPV creation via LocalVolume CR
+- The admin can select the devices detected by the `LocalVolumeDiscover` CR by creating a `LocalVolume` CR and explicitly specifying the devices into it.
+This https://github.com/openshift/local-storage-operator/blob/master/docs/deploy-with-olm.md doc explains how this can be achieved via explicitly providing the device paths.
+
+#### Workflow of LocalPV creation via AutoDetectVolume CR
+- The `AutoDetectVolume CR` will need a reference of `LocalVolumeDiscover` CR as the `LocalVolumeDiscover` CR holds the information of the discovered devices.
+Once the `AutoDetectVolume` is created, the controller will:
+configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
```

Comment:
What is based on certain directories? I thought we were only looking at devices. Or are you referring to the device path?

Reply:
yeah it refers to the device paths